airbyte-integrations/connectors/source-google-ads/source_google_ads/components.py

```diff
@@ -939,3 +894,144 @@ def _validate_query(self, query: str):
                 internal_message=f"The provided query is invalid: {query}. Please refer to the Google Ads API documentation for the correct syntax: https://developers.google.com/google-ads/api/fields/v20/overview and test validate your query using the Google Ads Query Builder: https://developers.google.com/google-ads/api/fields/v20/query_validator",
                 message=f"The provided query is invalid: {query}. Please refer to the Google Ads API documentation for the correct syntax: https://developers.google.com/google-ads/api/fields/v20/overview and test validate your query using the Google Ads Query Builder: https://developers.google.com/google-ads/api/fields/v20/query_validator",
             )
+
+
+@dataclass
+class StringParseState:
+    inside_string: bool = False
+    escape_next: bool = False
+    char_buffer: List[str] = field(default_factory=list)
+    last_read_key: Optional[str] = None
+
+
+@dataclass
+class ResultsArrayState:
+    within_results_array: bool = False
+    results_nesting_depth: int = 0
+    pending_results_colon: bool = False
+
+
+@dataclass
+class RowParseState:
+    collecting_row: bool = False
+    row_buffer: List[str] = field(default_factory=list)
+    row_nesting_depth: int = 0
+
+
+@dataclass
+class RowsStreamingDecoder(Decoder):
```

Comment:
nit: What does Row refers to here? Based on the code, I assume it is an entry within the `results` property, is that right? I'm not sure how the name could be better so can we document that at least?